### PR TITLE
Update Laravel 5 Doc for better Development

### PR DESCRIPTION
**Pull request by starvsion**
A more modern, and a more Fedora way to install Laravel, and Nginx is now more preferred for its performance.

### CODE REVIEW

File: start/sw/web-app/laravel5.md [hunk 1]
```diff
@@ -9,41 +9,49 @@ order: 4
...
-# Restart Apache
-$ sudo systemctl restart httpd
+# It will install all required extensions, perk of Fedora
+$ sudo dnf install composer
 ```
 
-#### Install Composer
-We will need composer to install laravel and it's dependencies
+#### Install Valet Linux
+We will need to set it up for easier development
```
Comment (yotamN) [inline start/sw/web-app/laravel5.md]: What is Valet? Maybe you could give a bit more info on what is it and why you need it instead of making the user jump to Laravel website.

File: start/sw/web-app/laravel5.md [hunk 2]
```diff
@@ -57,7 +65,7 @@ $ laravel new ProjectName
 ```bash
 $ cd ProjectName
 $ cp .env.example .env
-$ nano .env
+$ gedit .env
```
Comment (yotamN) [inline start/sw/web-app/laravel5.md]: Why would you change it? It seems a bit weird to change the editor. Nano is pretty standard for editing commands in tutorials.

File: start/sw/web-app/laravel5.md [hunk 3]
```diff
@@ -9,41 +9,49 @@ order: 4
...
+$ systemctl enable --now mysql
+# initialize MariaDB
+$ sudo mysql_secure_installation
+```
 
-#### Install PHP Extensions
+#### Install PHP and all its dependencies
+Follow the PHP installation instructions [here](/tech/languages/php/php-installation.html).
 
+##### Fedora 28 uses 7.2 by default
```
Comment (yotamN) [inline start/sw/web-app/laravel5.md]: Did you mean to make it a title or just bold?

Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Agreed, this does not llok quite well structured.

**Comment by pvalena** [comment]
@starvsion, PTAL at the requested changes.

File: start/sw/web-app/laravel5.md (hunk 3 above)
Comment (pvalena) [reply start/sw/web-app/laravel5.md]: https://guides.github.com/features/mastering-markdown/

Then please use a **different notation**, please.

File: start/sw/web-app/laravel5.md [hunk 4]
```diff
@@ -9,41 +9,51 @@ order: 4
...
+#### Install Nginx Web Server
+Follow the nginx installation instructions [here](/start/sw/web-app/nginx.html).
+
 
 #### Install MariaDB Relational Database Server
 Follow the MariaDB installation instructions [here](/tech/database/mariadb/about.html).
 
-#### Install PHP
-Follow the PHP installation instructions [here](/tech/languages/php/php-installation.html).
+Or you can install by :
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Install [what]? Why would you need to install mysql?

File: start/sw/web-app/laravel5.md [hunk 5]
```diff
@@ -9,41 +9,51 @@ order: 4
...
+#### Install PHP and all its dependencies
+Follow the PHP installation instructions [here](/tech/languages/php/php-installation.html).
 
 ```bash
-# Laravel required PHP extensions.
-$ sudo dnf install php php-common php-cli php-pdo php-mbstring php-zip php-xml
-# Restart Apache
-$ sudo systemctl restart httpd
+# It will install all required extensions, perk of Fedora
+$ sudo dnf install composer
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: At this point you haven't explained why you'd need to install composer either.

File: start/sw/web-app/laravel5.md [hunk 6]
```diff
@@ -9,41 +9,51 @@ order: 4
...
 ```
 
-#### Install Composer
-We will need composer to install laravel and it's dependencies
+#### Install Valet Linux
+We will need to set it up for easier development. Valet is original designed for MacOS by the Laravel Team; Valet Linux is a port to that. It setups up the neccesary environments for PHP Development, including a test domain. Please do not run valet as root. 
+For More Information, you can check them out here: 
+https://cpriego.github.io/valet-linux/
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Please make this a markdown link (named with what-the-page-is intead of url).

File: start/sw/web-app/laravel5.md [hunk 7]
```diff
@@ -9,41 +9,51 @@ order: 4
...
+# Install valet 
+$ composer global require cpriego/valet-linux
+# Initialize valet
+$ valet install
 ```
 
 #### Install laravel installer
 
 ```bash
 $ composer global require "laravel/installer"
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Ok, so now we're getting to work with laravel at last. The all previous were dependencies. Could you maybe denote that somehow in the beginning like:
```
## Dependencies for laravel on Fedora
 - mysql
 - composer
 - valet linux
```
Anyways, aren't those tools worthy enough to have their own standalone page? Like f.e.:
 - http://developer.fedorainfracloud.org/tools/vagrant/about.html
 - http://developer.fedorainfracloud.org/tools/docker/about.html

**Comment by pvalena** [comment]
Otherwise it LGTM, thanks!

File: start/sw/web-app/laravel5.md (hunk 7 above)
Comment (starvsion) [reply start/sw/web-app/laravel5.md]: vagrant and docker are vm-like technologies, and it is out of scope for setting up Laravel on Local Environments ...

File: start/sw/web-app/laravel5.md (hunk 4 above)
Comment (starvsion) [reply start/sw/web-app/laravel5.md]: You need to setup database when setting up Laravel, MySQL or MariaDB is just the most common one

Comment (starvsion) [reply start/sw/web-app/laravel5.md]: MySQL is just the most popular database used with PHP, and Laravel relies on database to fully function.

File: start/sw/web-app/laravel5.md (hunk 7 above)
Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Yes it is. I didn't mean to include the content in Laravel page, on the contrary, I was thinking of moving the content elsewhere and pointing to it from here. Anyway it was just a suggestion and the move can be done in later iterations.

My main point was, though, that there was no word on _Why_ you're installing those packages/tools. It would be nice to include the information, like stated above, in some `dependencies list`, or similar. As the developer can have already those installed, they can skip the step, etc.. (The list would be hyper-linked.)

File: start/sw/web-app/laravel5.md (hunk 4 above)
Comment (pvalena) [reply start/sw/web-app/laravel5.md]: > You need to setup database when setting up Laravel, MySQL or MariaDB is just the most common one
> MySQL is just the most popular database used with PHP, and Laravel relies on database to fully function.


Well, could you include that information in your guide then?

I'm not trying to argue, nor do I lack the information, I'm merely missing it in the content.

File: start/sw/web-app/laravel5.md [hunk 8]
```diff
@@ -9,46 +9,53 @@ order: 4
 
 Laravel is a high-level PHP Web framework.
 
-#### Install Apache Web Server
-Follow the Apache installation instructions [here](/start/sw/web-app/apache.html).
+#### Install Nginx Web Server
+Follow the nginx installation instructions [here](/start/sw/web-app/nginx.html).
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: This is just a matter of preference, right? Could you include both options then?

File: start/sw/web-app/laravel5.md [hunk 9]
```diff
@@ -57,7 +64,7 @@ $ laravel new ProjectName
 ```bash
 $ cd ProjectName
 $ cp .env.example .env
-$ nano .env
+$ gedit .env #or nano,vim,emacs
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Please move the comment to line above.
I'd say that there's no need to list the editors, but you can describe what you're about to do, e.g. `edit file XYZ`.

_Note: Using one or other is just a preference, and AFAIK `gedit` is in default installation. So it's fine with me._

File: start/sw/web-app/laravel5.md [hunk 10]
```diff
@@ -71,7 +78,27 @@ DB_PASSWORD=Chur1
...
+
+###### if ssl is desired
+
+```bash
+$ valet secure
+# Now visit https://DOMAIN.test/
+
+```
+
+###### if need a callback URL/share work
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Looking at the <hX> elements of markdown, I can see you're only skipping the odd ones(1,3,5...). Why wouldn't you start from `# Header 1` and continue with `## Header 2`, `### Header 3`...?

 - https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

File: start/sw/web-app/laravel5.md [hunk 11]
```diff
@@ -9,46 +9,53 @@ order: 4
...
 ```
 
-Restart Terminal. Now you can create a laravel project with the commands 
+#### Install laravel installer
 
 ```bash
+# Install Laravel Installer
+$ composer global require "laravel/installer"
+# use Laravel Installer to Install Laravel
 $ laravel new ProjectName
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: The comment above seems quite confusing. Are you really installing laravel here with this command?

File: start/sw/web-app/laravel5.md [hunk 12]
```diff
@@ -9,46 +9,53 @@ order: 4
...
+# initialize MariaDB
+$ sudo mysql_secure_installation
 ```
 
-#### Install Composer
-We will need composer to install laravel and it's dependencies
+#### Install PHP and all its dependencies
+Follow the PHP installation instructions [here](/tech/languages/php/php-installation.html).
 
+OR, you can rip the benefit of using Fedora, and install composer directly instead. This will install PHP, and all the required extensions/libraries needed with dnf (And of course, updates with dnf).
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: In that case, I'd say you can just state that php will be pulled in with composer and there's no need to follow the instructions (just point to them in case user needs them for some reason).

Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Also please omit the part about updates(specifically `needed with dnf (And of course, updates with dnf).`). We try to keep it simple.

File: start/sw/web-app/laravel5.md (hunk 4 above)
Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Also please omit the `or you can install by`. We try not to duplicate the content.

File: start/sw/web-app/laravel5.md [hunk 13]
```diff
@@ -9,46 +9,53 @@ order: 4
...
-$ sudo mv composer.phar /usr/local/bin/composer
-$ chmod +x /usr/local/bin/composer
-$ composer -V
+# It will install all required extensions, perk of Fedora
+$ sudo dnf install composer
 ```
 
-#### Install laravel installer
+#### Install Valet Linux
+We will need to set it up for easier development. Valet is original designed for MacOS by the Laravel Team; Valet Linux is a port to that. It setups up the neccesary environments for PHP Development, including a test domain. Please do not run valet as root.
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: > Valet is original designed for MacOS by the Laravel Team; Valet Linux is a port to that. 

Please skip this history part. It's more imporant to state *what it does*.

**Comment by pvalena** [comment]
Thanks a lot for your work.

File: start/sw/web-app/laravel5.md (hunk 13 above)
Comment (starvsion) [reply start/sw/web-app/laravel5.md]: Linux Valet uses Laravel Valet commands and most of its codebase, so it's a good thing to mention

Comment (pvalena) [reply start/sw/web-app/laravel5.md]: I still do not see the point for someone who's never worked with Laravel, nor *Valet. Please keep it short for the sake of *quickstart*. Everything should be written like a short `TL;DR`, or the user will skip it.

Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Also, related to our [contributing guide](http://developer-portal.github.io/contributing/write-general-writing-hints.html), I'd reword the following sentence to
> Setup environment for PHP development...

Also, I do not see a point in:

> Please do not run valet as root.

As any developer knows not to do that and this is not a place to *teach* them. (Nor is it a documentation.)

**Comment by pvalena** [comment]
Please make the necessary changes(see [my comments](https://github.com/developer-portal/content/pull/292/files)), unfortunately it's not possible to merge it in current state.

File: start/sw/web-app/laravel5.md (hunk 8 above)
Comment (starvsion) [reply start/sw/web-app/laravel5.md]: Actually no, Laravel Valet uses Nginx

Comment (pvalena) [reply start/sw/web-app/laravel5.md]: Ok, then please mention this in the quickstart.

File: start/sw/web-app/laravel5.md [hunk 14]
```diff
@@ -40,6 +44,8 @@ We will need to set it up for easier development. Valet is original designed for
 For More Information, you can check them out here: 
 [Valet Linux](https://cpriego.github.io/valet-linux/)  and  [Laravel Valet](https://laravel.com/docs/5.7/valet)
 
+You can also setup Laravel using [Vagrant](http://developer.fedorainfracloud.org/tools/vagrant/about.html), [Docker](http://developer.fedorainfracloud.org/tools/docker/about.html).
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Please remove the website name ``http://developer.fedorainfracloud.org`` from the link.

Note that you're linking to the development instance!

File: start/sw/web-app/laravel5.md [hunk 15]
```diff
@@ -64,7 +71,7 @@ $ laravel new ProjectName
 ```bash
 $ cd ProjectName
 $ cp .env.example .env
-$ gedit .env #or nano,vim,emacs
+$ gedit .env #Editing your environment file
```
Comment (pvalena) [inline start/sw/web-app/laravel5.md]: Please move the comment above the commands.

**Comment by pvalena** [comment]
Great, just a last bit of polishing and we can merge it.

File: start/sw/web-app/laravel5.md (hunk 4 above)
Comment (pvalena) [reply start/sw/web-app/laravel5.md]: @starvsion, can you deduplicate it? Or shall I?

Comment (starvsion) [reply start/sw/web-app/laravel5.md]: > @starvsion, can you deduplicate it? Or shall I?

you do it ...

**Comment by pvalena** [comment]
Looks great, thanks!
